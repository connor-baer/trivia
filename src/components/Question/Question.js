import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import { Heading } from '@sumup/circuit-ui';

const baseStyles = ({ theme }) => css`
  font-weight: ${theme.fontWeight.regular};
`;

const Question = styled(Heading)(baseStyles);

Question.propTypes = {
  children: PropTypes.string.isRequired
};

/**
 * @component
 */
export default Question;
