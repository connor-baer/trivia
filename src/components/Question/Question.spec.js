import React from 'react';

import Question from './Question';

const defaultProps = {
  children: 'What is your name?'
};

describe('Question', () => {
  it('should render with default styles', () => {
    const { container } = render(<Question {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Question {...defaultProps} />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
