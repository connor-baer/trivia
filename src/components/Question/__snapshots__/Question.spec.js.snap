// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render with default styles 1`] = `
.emotion-0 {
  font-weight: 700;
  margin-bottom: 24px;
  font-size: 24px;
  line-height: 32px;
  font-weight: 400;
}

@media (min-width:480px) {
  .emotion-0 {
    font-size: 28px;
    line-height: 32px;
  }
}

<h2
  class="emotion-0 emotion-1"
>
  What is your name?
</h2>
`;
